benches/pcs.rs

```diff
@@ -0,0 +1,181 @@
+          config = Criterion::default().warm_up_time(Duration::from_millis(3000)).with_profiler(pprof::criterion::PProfProfiler::new(100, pprof::criterion::Output::Flamegraph(None)));
+          targets = bench_pcs
+    }
+  } else {
+    criterion_group! {
+          name = pcs;
+          config = Criterion::default().warm_up_time(Duration::from_millis(3000));
+          targets = bench_pcs
+    }
+  }
+}
+
+criterion_main!(pcs);
+
+const TEST_ELL: [usize; 11] = [10, 11, 12, 23, 14, 15, 16, 17, 18, 19, 20];
+
+struct BenchAssests<E: Engine, EE: EvaluationEngineTrait<E>> {
+  poly: MultilinearPolynomial<<E as Engine>::Scalar>,
+  point: Vec<<E as Engine>::Scalar>,
+  eval: <E as Engine>::Scalar,
+  ck: <<E as Engine>::CE as CommitmentEngineTrait<E>>::CommitmentKey,
+  commitment: <<E as Engine>::CE as CommitmentEngineTrait<E>>::Commitment,
+  prover_key: <EE as EvaluationEngineTrait<E>>::ProverKey,
+  verifier_key: <EE as EvaluationEngineTrait<E>>::VerifierKey,
+  proof: Option<<EE as EvaluationEngineTrait<E>>::EvaluationArgument>,
+}
+
+// Macro to generate benchmark code for multiple engine and evaluation engine types
+macro_rules! benchmark_engines {
+    ($ell:expr, $rng:expr, $group:expr, $internal_fn:expr, $( ($engine:ty, $eval_engine:ty, $engine_name:expr) ),*) => {
+        $(
+            let mut assets = deterministic_assets::<$engine, $eval_engine, StdRng>($ell, &mut $rng);
+            $group.bench_with_input(BenchmarkId::new($engine_name, $ell), &$ell, |b, _| {
+                $internal_fn(b, &mut assets);
+            });
+        )*
+    };
+}
+
+fn bench_pcs(c: &mut Criterion) {
+  for &ell in TEST_ELL.iter() {
+    let mut rng = rand::rngs::StdRng::seed_from_u64(ell as u64);
+
+    // Proving group
+    {
+      let mut proving_group = c.benchmark_group("PCS-Proving");
+      proving_group
+        .sampling_mode(SamplingMode::Flat)
+        .sample_size(10);
+
+      benchmark_engines!(
+          ell,
+          rng,
+          proving_group,
+          bench_pcs_proving_internal,
+          (GrumpkinEngine, IPAEvaluationEngine<GrumpkinEngine>, "IPA"),
+          (Bn256EngineKZG, MLEvaluationEngine<Bn256, Bn256EngineKZG>, "MLKZG"),
+          (Bn256EngineZM, ZMPCS<Bn256, Bn256EngineZM>, "ZM")
+      );
+
+      proving_group.finish();
+    }
+
+    // Verifying group
+    {
+      let mut verifying_group = c.benchmark_group("PCS-Verifying");
+      verifying_group
+        .sampling_mode(SamplingMode::Flat)
+        .sample_size(10);
+
+      benchmark_engines!(
+          ell,
+          rng,
+          verifying_group,
+          bench_pcs_verifying_internal,
+          (GrumpkinEngine, IPAEvaluationEngine<GrumpkinEngine>, "IPA"),
+          (Bn256EngineKZG, MLEvaluationEngine<Bn256, Bn256EngineKZG>, "MLKZG"),
+          (Bn256EngineZM, ZMPCS<Bn256, Bn256EngineZM>, "ZM")
+      );
+
+      verifying_group.finish();
+    }
+  }
+}
+
+fn deterministic_assets<E: Engine, EE: EvaluationEngineTrait<E>, R: CryptoRng + RngCore>(
```

Comment:
You're right that the naming here is off. I specified it at `deterministic...` because if was a pre-determined polynomial size used to generate the assets. I've the implementation to be over `BenchAssets` and the name of the method to `from_num_vars`. should be more explicit this way.